kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/FilterHarness.java

```diff
@@ -91,7 +91,8 @@ protected void buildChannel(Filter... filters) {
                     return d2;
                 })) // reverses order
                 .stream()
-                .map(f -> new FilterHandler(getOnlyElement(FilterAndInvoker.build(f.getClass().getSimpleName(), f)), timeoutMs, null, testVirtualCluster, inboundChannel))
+                .map(f -> new FilterHandler(getOnlyElement(FilterAndInvoker.build(f.getClass().getSimpleName(), f)), timeoutMs, null, testVirtualCluster, inboundChannel,
+                        new ClientSaslContextImpl()))
```

Comment:
To get close to production like behaviour, the same CSC should be shared by all the filters

Reply:
Good catch, thanks!